### PR TITLE
Move HighlightActions from Actions to SharedUx

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Describe the change
We are moving HighlightAction(s) to SharedUx from Actions so that Actions reduces its UI dependencies/footprint. Most of the files are changing a reference from Actions to SharedUx. We still need to find better names for HighlightAction since Highlight may be repetitive (the parent folder is called Highlighting) and it's not strictly in 'Action' in the same sense as the files in our Actions project.

#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

Note - After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

**Comment by jalkire** [comment]
Are we planning on moving the highlighters themselves out of the runtime into SharedUx, since they are actually experienced by the user? It seems like where they belong, especially after this change.

File: src/AccessibilityInsights/MainWindowHelpers/Initialize.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@
 using System.Windows.Shell;
 using System.Globalization;
 using AccessibilityInsights.Desktop.Telemetry;
```
Comment (DaveTryon) [inline src/AccessibilityInsights/MainWindowHelpers/Initialize.cs]: >AccessibilityInsights [](start = 6, length = 21)

Sort using blocks alphabetically? Looks like Actions was originally misplaced

File: src/AccessibilityInsights.SharedUx/Highlighting/HighlightOverlayAction.cs [hunk 2]
```diff
@@ -5,22 +5,20 @@
...
 using System.Windows.Media;
 using AccessibilityInsights.Core.Bases;
-using AccessibilityInsights.Actions.Attributes;
-using AccessibilityInsights.Actions.Enums;
 using System.Diagnostics;
 using System.Runtime.InteropServices;
 using UIAutomationClient;
 using System.Windows.Controls;
 using AccessibilityInsights.Win32;
+using AccessibilityInsights.Actions;
```
[the remark this answers is not in the document]
Comment (DaveTryon) [reply src/AccessibilityInsights.SharedUx/Highlighting/HighlightOverlayAction.cs]: Once the runtime namespace is converted, we'll probably want to make a pass where we sort all of the using blocks. I'm not sure how much effort we want to invest in fixing it now when we'll be revisiting it again so soon

---
In reply to: [270230355](https://github.com/Microsoft/accessibility-insights-windows/pull/232#discussion_r270230355) [](ancestors = 270230355)

**Comment by DaveTryon** [comment]
@karanbirsingh, what's the plan with FabricIcon? It's still in DesktopUI, but the git-issues PR creates a copy in the CommonUxControls assembly. Will Highlighter eventually pull it from CommonUxControls?

File: src/AccessibilityInsights/MainWindowHelpers/StateMachine.cs [hunk 3]
```diff
@@ -138,7 +139,7 @@ public void HandleBackToSelectingState()
             }
 
             // clear selected element highliter
-            HighlightAction.GetInstance(AccessibilityInsights.Actions.Enums.HighlighterType.Selected).Clear();
+            HighlightAction.GetInstance(AccessibilityInsights.SharedUx.Enums.HighlighterType.Selected).Clear();
```
Comment (neocln) [inline src/AccessibilityInsights/MainWindowHelpers/StateMachine.cs]: >AccessibilityInsights.SharedUx.Enums. [](start = 40, length = 37)

how about using "using" statement here and below?

**Comment by karanbirsingh** [comment]
> @karanbirsingh, what's the plan with FabricIcon? It's still in DesktopUI, but the git-issues PR creates a copy in the CommonUxControls assembly. Will Highlighter eventually pull it from CommonUxControls?

I think that is a good idea. It sounds like the plan is to move Highlighter-related classes to SharedUx and remove DesktopUI entirely.

**Comment by karanbirsingh** [comment]
> Are we planning on moving the highlighters themselves out of the runtime into SharedUx, since they are actually experienced by the user? It seems like where they belong, especially after this change.

Yes, thanks for suggesting/mentioning this

File: src/AccessibilityInsights.SharedUx/Highlighting/HighlightOverlayAction.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (karanbirsingh) [reply src/AccessibilityInsights.SharedUx/Highlighting/HighlightOverlayAction.cs]: I have sorted the usings in this and other Highlighting files. Can help circle back later too for other files as Dave suggests

File: src/AccessibilityInsights.SharedUx/ActionViews/ReturnA11yElementsView.xaml.cs [hunk 4]
```diff
@@ -56,15 +57,15 @@ public static bool HighlightVisibility
             }
             set
             {
-                var ha = HighlightOverlayAction.GetDefaultInstance();
-                var sha = HighlightAction.GetDefaultInstance();
+                var ha = ClearOverlayDriver.GetDefaultInstance();
```
Comment (DaveTryon) [inline src/AccessibilityInsights.SharedUx/ActionViews/ReturnA11yElementsView.xaml.cs]: >ha [](start = 20, length = 2)

These local variable names seem kind of weird for the renamed types

File: src/AccessibilityInsights.SharedUx/ActionViews/ReturnA11yElementsView.xaml.cs [hunk 5]
```diff
@@ -101,10 +102,10 @@ public void MarkElement(A11yElement ele)
         {
             lock (this)
             {
-                if (HighlightAction.GetDefaultInstance().IsEnabled)
+                if (HollowHighlightDriver.GetDefaultInstance().IsEnabled)
                 {
-                    var ha = HighlightOverlayAction.GetDefaultInstance();
```
Comment (DaveTryon) [inline src/AccessibilityInsights.SharedUx/ActionViews/ReturnA11yElementsView.xaml.cs]: >ha [](start = 24, length = 2)

Again, weird local names for the new type names

File: src/AccessibilityInsights/MainWindowHelpers/StateMachine.cs [hunk 6]
```diff
@@ -150,7 +148,7 @@ public void HandleBackToSelectingState()
 
             // turn highlighter on once you get back to selection mode. 
             SetHighlightBtnState(true);
-            var ha = HighlightAction.GetDefaultInstance();
```
Comment (DaveTryon) [inline src/AccessibilityInsights/MainWindowHelpers/StateMachine.cs]: >ha [](start = 16, length = 2)

Weird local name

File: src/AccessibilityInsights.SharedUx/ActionViews/ReturnA11yElementsView.xaml.cs (hunk 4 above)
Comment (karanbirsingh) [reply src/AccessibilityInsights.SharedUx/ActionViews/ReturnA11yElementsView.xaml.cs]: good catch, I did an auto-rename and will update these types

File: src/AccessibilityInsights/Modes/SnapshotModeControl.xaml.cs [hunk 7]
```diff
@@ -170,7 +171,7 @@ public async Task SetElement(Guid ecId)
                                 Application.Current.MainWindow.WindowStyle = WindowStyle.SingleBorderWindow;
                             })).Wait();
                         }
-                        var ha = HighlightImageAction.GetDefaultInstance();
```
Comment (DaveTryon) [inline src/AccessibilityInsights/Modes/SnapshotModeControl.xaml.cs]: >ha [](start = 28, length = 2)

Rename local to match new type?

File: src/AccessibilityInsights/MainWindow.xaml.cs [hunk 8]
```diff
@@ -99,7 +100,7 @@ public string AutomationPropertiesNameInspect
             {
                 return string.Format(CultureInfo.InvariantCulture, Properties.Resources.MainWindow_AutomationPropertiesName_0_1_2, Properties.Resources.btnInspectAutomationPropertiesName, this.CurrentPage.ToString(), Properties.Resources.ModeIsEnabled);
             }
-        }
+
```
Comment (DaveTryon) [inline src/AccessibilityInsights/MainWindow.xaml.cs]: Looks like the closing brace got lost in the merge

**Pull request merged**